components/adblock_rust_ffi/src/lib.rs

```diff
@@ -457,18 +458,35 @@ pub unsafe extern "C" fn engine_hidden_class_id_selectors(
     exceptions: *const *const c_char,
     exceptions_size: size_t,
 ) -> *mut c_char {
-    let classes = std::slice::from_raw_parts(classes, classes_size);
-    let classes: Vec<String> = (0..classes_size)
-        .map(|index| CStr::from_ptr(classes[index]).to_str().unwrap().to_owned())
-        .collect();
-    let ids = std::slice::from_raw_parts(ids, ids_size);
-    let ids: Vec<String> = (0..ids_size)
-        .map(|index| CStr::from_ptr(ids[index]).to_str().unwrap().to_owned())
-        .collect();
-    let exceptions = std::slice::from_raw_parts(exceptions, exceptions_size);
-    let exceptions: std::collections::HashSet<String> = (0..exceptions_size)
-        .map(|index| CStr::from_ptr(exceptions[index]).to_str().unwrap().to_owned())
-        .collect();
+    // Note checks for `size == 0` - `std::vector<T>::data()`'s return value when empty is
+    // undefined behavior and should never be used.
+    let classes = if classes_size == 0 {
```

Comment:
@cuba just a note, I changed it to check for _zero size_ rather than a nullptr, just in case the undefined behavior manifests in different ways (e.g. returning some totally random memory address)

Reply:
Good call